docs/milestone3/milestone3_walkthrough.md

```diff
@@ -0,0 +1,137 @@
+
+## Environment Setup
+
+Create a set of environment files with your AWS configuration.
+First, create a file called `aws-credentials.env` in the root of the project and add the following config, replacing values as necessary
+
+```bash
+AWS_ACCESS_KEY_ID=<AWS_ACCESS_KEY_ID>
+AWS_SECRET_ACCESS_KEY=<AWS_SECRET_ACCESS_KEY>
+AWS_REGION=<AWS_REGION>
+```
+
+Next, create a file called `controller-config.env` in the root of the project and add the following config, replacing values as necessary. For example, if your Hosted zone is for the domain `apps.example.com`, use that as the `ZONE_ROOT_DOMAIN`. This will mean your Application can use a hostname of `myapp.apps.example.com`
+
+```bash
+ZONE_ROOT_DOMAIN=<my.hosted.zone.domain.name>
+AWS_DNS_PUBLIC_ZONE_ID=<my.hosted.zone.id>
+```
+
+Start the local clusters (1 control plane & 2 data plane workload clusters)
+
+```bash
+MCTC_WORKLOAD_CLUSTERS_COUNT=2 make local-setup
+```
+
+Run the controller in the control plane with the configuration from the env files.
+
+```bash
+export KUBECONFIG=./tmp/kubeconfigs/mctc-control-plane.kubeconfig
+(export $(cat ./controller-config.env | xargs) && export $(cat ./aws-credentials.env | xargs) && make build-controller install run-controller)
+```
+
+In a new terminal, create the multi cluster Gateway Class and tenant namespace
+
+```bash
+export KUBECONFIG=./tmp/kubeconfigs/mctc-control-plane.kubeconfig
+kubectl apply -f config/samples/gatewayclass.yaml
+kubectl create namespace mctc-tenant
+```
+
+Start the syncer component in both workload clusters
+
+```bash
+export KUBECONFIG=./tmp/kubeconfigs/mctc-workload-1.kubeconfig
+make kind-load-syncer
+make deploy-syncer
+kubectl wait --for=condition=Available deployment sync-agent -n mctc-system
+
+export KUBECONFIG=./tmp/kubeconfigs/mctc-workload-2.kubeconfig
+kind load docker-image syncer:latest --name mctc-workload-2 --nodes mctc-workload-2-control-plane
+make deploy-syncer
+kubectl wait --for=condition=Available deployment sync-agent -n mctc-system
+```
+
+## Gateway Setup
+
+Export the hostname you want to use for you Application, and copy the example gateway.
+For example, if you set your `ZONE_ROOT_DOMAIN` as `example.com`, you could set `MYAPP_HOST` to `myapp.example.com` below to use it as your Application host.
+
+```bash
+export MYAPP_HOST=<myapp.host>
+envsubst < docs/milestone3/milestone3_gateway.yaml > gateway.yaml
+```
+
+Create the Gateway in the control plane.
+
+```bash
+export KUBECONFIG=./tmp/kubeconfigs/mctc-control-plane.kubeconfig
+kubectl apply -n mctc-tenant -f ./gateway.yaml
+```
+
+Create a Placement to place the Gateway on both workload clusters
+
+```bash
+export KUBECONFIG=./tmp/kubeconfigs/mctc-control-plane.kubeconfig
+kubectl apply -n mctc-tenant -f docs/milestone3/milestone3_placement.yaml
+```
+
+The Gateway should now be running in both workload clusters
+
+```bash
+export KUBECONFIG=./tmp/kubeconfigs/mctc-workload-1.kubeconfig
+kubectl get gateway example-gateway -n mctc-downstream
```

Comment:
the namespace is mctc-downstream is throwing a error i think it should be mctc-system

Reply:
Yes. I've pushed a couple updates after pulling in the syncer status fix from main